integration/src/main/java/scripts/utils/ClientTestUtils.java

```diff
@@ -338,6 +341,76 @@ public static <T> T getWithRetryOnException(
     return result;
   }
 
+  /**
+   * Get a result from a call that might throw an exception. If the supplier finishes, the result is
+   * returned. If the supplier continues to throw, when totalDuration has elapsed, this method will
+   * throw that exception.
+   *
+   * @param predicate - if evaluated true, then get the result; Otherwise, retry.
+   * @param supplier - code returning the result or throwing an exception
+   * @param totalDuration - total amount of time to retry
+   * @param initialSleepDuration - initial amount of time to sleep between retries
+   * @param factorIncrease - factor to increase the sleep time. The formula is: newSleepDuration =
+   *     sleepDuration + (factorIncrease * sleepDuration) The default of 0.0 results in a fixed
+   *     wait.
+   * @param sleepDurationMax = the maximum duration to expand the sleep time.
+   * @param <T> - type of result
+   * @return - result from supplier, if no exception
+   * @throws InterruptedException if the sleep is interrupted
+   */
+  public static <T> T getWithRetry(
+      Predicate<T> predicate,
+      SupplierWithException<T> supplier,
+      Duration totalDuration,
+      Duration initialSleepDuration,
+      double factorIncrease,
+      Duration sleepDurationMax)
+      throws Exception {
+
+    T result;
+    Instant endTime = Instant.now().plus(totalDuration);
+    Duration sleepDuration = initialSleepDuration;
+
+    while (true) {
+      result = supplier.get();
+      if (predicate.test(result)) {
+        break;
+      } else {
+        // If we are out of time
+        if (Instant.now().isAfter(endTime)) {
+          throw new Exception();
```

Comment:
this should be a more specific exception, and contain a message indicating what happened (e.g. "timed out during getWithRetry")